jbang fails to download jbang --verbose "https://gluonhq.com/products/scene-builder/thanks/?dl=https://download2.gluonhq.com/scenebuilder/21.0.0/scenebuilder-kit-21.0.0.jar"

scenebuilder url at  "https://gluonhq.com/products/scene-builder/thanks/?dl=https://download2.gluonhq.com/scenebuilder/21.0.0/scenebuilder-kit-21.0.0.jar" fails with:

```
[jbang] [0:235] Type in your choice and hit enter. Will automatically select option (0) after 30 seconds.
1
[jbang] [1:931] Requesting HTTP GET https://gluonhq.com/products/scene-builder/thanks/?dl=https://download2.gluonhq.com/scenebuilder/21.0.0/scenebuilder-kit-21.0.0.jarmain.java
[jbang] [1:931] Headers {User-Agent=[JBang/0.114.0 (Mac OS X/14.2.1/aarch64) Java/21.0.1/GraalVM Community]}
[jbang] [2:827] [ERROR] Range [104, 50) out of bounds for length 140
java.lang.StringIndexOutOfBoundsException: Range [104, 50) out of bounds for length 140
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55)
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112)
	at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349)
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4861)
	at java.base/java.lang.String.substring(String.java:2830)
	at dev.jbang.util.Util.extractFileName(Util.java:1069)
	at dev.jbang.util.Util.access$300(Util.java:75)
	at dev.jbang.util.Util$ResultHandler.lambda$downloadTo$2(Util.java:948)
	at dev.jbang.util.Util$ResultHandler.lambda$downloadToTempDir$3(Util.java:980)
	at dev.jbang.util.Util$ResultHandler.lambda$throwOnError$1(Util.java:941)
	at dev.jbang.util.Util$ResultHandler.lambda$redirects$0(Util.java:894)
	at dev.jbang.util.Util.connect(Util.java:814)
	at dev.jbang.util.Util.downloadFileAndCache(Util.java:743)
	at dev.jbang.util.Util.downloadAndCacheFile(Util.java:694)
	at dev.jbang.source.resolvers.RemoteResourceResolver.fetchScriptFromUntrustedURL(RemoteResourceResolver.java:91)
	at dev.jbang.source.resolvers.RemoteResourceResolver.resolve(RemoteResourceResolver.java:42)
	at dev.jbang.source.resolvers.CombinedResourceResolver.lambda$resolve$0(CombinedResourceResolver.java:27)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1034)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
	at dev.jbang.source.resolvers.CombinedResourceResolver.resolve(CombinedResourceResolver.java:29)
	at dev.jbang.source.resolvers.AliasResourceResolver.resolve(AliasResourceResolver.java:37)
	at dev.jbang.source.ResourceResolver.resolve(ResourceResolver.java:52)
	at dev.jbang.source.ProjectBuilder.resolveChecked(ProjectBuilder.java:229)
	at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:218)
	at dev.jbang.cli.Run.doCall(Run.java:73)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:145)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:21)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at dev.jbang.cli.JBang$3.handle(JBang.java:148)
	at dev.jbang.cli.JBang$3.handle(JBang.java:143)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:20
```

--- a/src/main/java/dev/jbang/util/Util.java
+++ b/src/main/java/dev/jbang/util/Util.java
@@ -1066,7 +1066,12 @@ public class Util {
 				if (isBlankString(fileName)) {
 					// extracts file name from URL if nothing found
 					int p = fileURL.indexOf("?");
-					fileName = fileURL.substring(fileURL.lastIndexOf("/") + 1, p > 0 ? p : fileURL.length());
+					// Strip parameters from the URL (if any)
+					String simpleUrl = (p > 0) ? fileURL.substring(0, p) : fileURL;
+					while (simpleUrl.endsWith("/")) {
+						simpleUrl = simpleUrl.substring(0, simpleUrl.length() - 1);
+					}
+					fileName = simpleUrl.substring(simpleUrl.lastIndexOf("/") + 1);
 				}
 			}
 		} else {

--- a/src/test/java/dev/jbang/util/TestUtilDownloads.java
+++ b/src/test/java/dev/jbang/util/TestUtilDownloads.java
@@ -404,6 +404,22 @@ public class TestUtilDownloads extends BaseTest {
 		});
 	}
 
+	@Test
+	void testReqUrlWithParams(WireMockRuntimeInfo wmri) throws IOException {
+		stubFor(get(urlEqualTo("/test.txt?path=foo/bar"))
+															.andMatching(withoutHeader("If-None-Match"))
+															.andMatching(withoutHeader("If-Modified-Since"))
+															.willReturn(aResponse()
+																					.withHeader("Content-Type",
+																							"text/plain")
+																					.withBody("test")));
+
+		String url = wmri.getHttpBaseUrl() + "/test.txt?path=foo/bar";
+		Path file = Util.downloadAndCacheFile(url);
+		assertThat(file.toFile(), anExistingFile());
+		assertThat(Util.readString(file), is("test"));
+	}
+
 	private static ValueMatcher<Request> withoutHeader(String hdr) {
 		return req -> MatchResult.of(!req.containsHeader(hdr));
 	}

